### PR TITLE
add bleu script

### PR DESCRIPTION
**Pull request by mhn226**
This PR introduces a BLEU measuring script, which exploits sacrebleu (https://github.com/mjpost/sacrebleu) to measure BLEU scores for Speech Translation tasks. It is therefore necessary to install sacrebleu for the script to work:

`pip install sacrebleu`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,13 +1,14 @@
 -r lint-requirements.txt
+huggingface_hub>=0.0.6
 hyperpyyaml>=0.0.1
 joblib>=0.14.1
 numpy>=1.17.0
 packaging
 pre-commit>=2.3.0
+sacrebleu>=1.5.1
```
Comment (TParcollet) [inline requirements.txt:8]: Please remove it from the general requirements.txt and instead try to see if it's installed when importing. If not, then print a message to install it ?

File: speechbrain/utils/bleu.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
+import sacrebleu
+from speechbrain.dataio.dataio import merge_char, split_word
+from speechbrain.utils.metric_stats import MetricStats
+
+
+def merge_words(sequences):
```
Comment (TParcollet) [inline speechbrain/utils/bleu.py:6]: Could you add a small description of what this function does ? And why we need it ?

File: speechbrain/utils/bleu.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    results = []
+    for seq in sequences:
+        words = " ".join(seq)
+        results.append(words)
+    return results
+
+
+def detokenize_batch(detokenizer, sentences):
+    """
+    detokenizer: object
```
Comment (TParcollet) [inline speechbrain/utils/bleu.py:16]: Short description of what the function does (even if it's clear from the name, it's better to explain it quickly for the users")

File: speechbrain/utils/bleu.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    detok_sentences = []
+    for sentence in sentences:
+        sentence = detokenizer.detokenize(sentence)
+        detok_sentences.append(sentence)
+
+    return detok_sentences
+
+
+class BLEUStats(MetricStats):
+    """A class for tracking BLEU.
```
Comment (TParcollet) [inline speechbrain/utils/bleu.py:30]: Docstring ok. Maybe a reference to a paper on BLEU ?

File: speechbrain/utils/bleu.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        this token in between, and then the whole sequence is split.
+    Example
+    -------
+    >>> bleu = BLEUStats()
+    >>> i2l = {0: 'a', 1: 'b'}
+    >>> bleu.append(
+    ...     ids=['utterance1'],
+    ...     predict=[[0, 1, 1]],
+    ...     target=[[0, 1, 0]],
+    ...     ind2lab=lambda batch: [[i2l[int(x)] for x in seq] for seq in batch],
```
Comment (TParcollet) [inline speechbrain/utils/bleu.py:57]: This line might be confusing. Consider doing a line before and explaining it briefly.

File: speechbrain/utils/bleu.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        self.space_token = space_token
+
+        self.predicts = []
+        self.targets = []
+
+    def append(
+        self, ids, predict, target, ind2lab=None,
+    ):
+        """Add stats to the relevant containers.
+        * See MetricStats.append()
```
Comment (TParcollet) [inline speechbrain/utils/bleu.py:85]: Link to this doc.

File: speechbrain/utils/bleu.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        Arguments
+        ---------
+        ids : list
+            List of ids corresponding to utterances.
+        predict : torch.tensor
+            A predicted output, for comparison with the target output
+        target : torch.tensor
+            The correct reference output, for comparison with the prediction.
```
Comment (Gastron) [inline speechbrain/utils/bleu.py:93]: This doc and the implementation seem to assume one reference - but BLEU was specifically designed to handle multiple references, so we should support that as well.

**Comment by TParcollet** [comment]
__main__ - Average batchsize: 7.9523809523809526 -> Wouldn't this information better be in the log.log only ?

**Comment by TParcollet** [comment]
LGTM Thanks @mhn226 !

**Pull request merged**